packages/annotations/src/store/test/reducer.js

```diff
@@ -0,0 +1,163 @@
+import { annotations } from '../reducer';
+
+describe( 'annotations', () => {
+	const initialState = { all: [], byBlockClientId: {} };
+
+	it( 'returns all annotations and annotation IDs per block', () => {
+		const state = annotations( undefined, {} );
+
+		expect( state ).toEqual( { all: [], byBlockClientId: {} } );
+	} );
+
+	it( 'returns a state with an annotation that has been added', () => {
+		const state = annotations( undefined, {
+			type: 'ANNOTATION_ADD',
+			id: 'annotationId',
+			blockClientId: 'blockClientId',
+			richTextIdentifier: 'identifier',
+			source: 'default',
+			selector: 'block',
+		} );
+
+		expect( state ).toEqual( {
+			all: [
+				{
+					id: 'annotationId',
+					blockClientId: 'blockClientId',
+					richTextIdentifier: 'identifier',
+					source: 'default',
+					selector: 'block',
+				},
+			],
+			byBlockClientId: {
+				blockClientId: [ 'annotationId' ],
+			},
+		} );
+	} );
+
+	it( 'allows an annotation to be removed', () => {
+		const state = annotations( {
```

Comment:
A good enhancement to these tests would be to apply `deepFreeze` on the original states. (also need a dev dependency)